### DISCUSSION TITLE
Keda for Azure service bus - Metrics API (on dead letter)

### DISCUSSION THREAD

**Discussion by benfrankvbf** [kedacore/keda#4447] [Q&A / Need Help]
Hi,

We are trying to utilize keda scaledobject for autoscaling our deployments, triggered by the result of Metrics API (Azure). Azure API 'management.azure.com' is using bearer authentication and its expiring every one hour (Attached the code below). Its not practical to renew the token every one hour from our side. Do we have any better way of scaling the deployments based on the count of dead letter for the queue in Azure service bus.

  triggers:
    - type: metrics-api
      metadata:
        targetValue: "1"
        url: "https://management.azure.com/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/Microsoft.ServiceBus/namespaces/{namespace}/queues/{queue_name}?api-version=2021-11-01"
        valueLocation: 'properties.countDetails.deadLetterMessageCount'

We took a look at the Azure service bus trigger provided by keda, which also considers only the active message count and not the dead letter.

Thanks in Advance for any other suggestions to scale based on dead letter count.

**Comment by JorTurFer** [comment]
@tomkerkhove , do you know if that information is accessible somehow? Maybe using azure monitor?

**Comment by Nwayes** [comment]
Hello,
I am also intrested by this discussion.
How to do autoscaling with keda on dead letter queues for azure service bus ?
Thank you
